xbmc/pvr/addons/PVRClient.cpp

```diff
@@ -1647,299 +1657,313 @@ int CPVRClient::GetPriority() const
+  HandleAddonCallback(__func__, kodiInstance, [&](CPVRClient* client) {
+    // update all channel groups in the next iteration of the pvrmanager's main loop
+    CServiceBroker::GetPVRManager().TriggerChannelGroupsUpdate();
+  });
 }
 
 void CPVRClient::cb_trigger_epg_update(void* kodiInstance, unsigned int iChannelUid)
 {
-  CPVRClient* client = static_cast<CPVRClient*>(kodiInstance);
-  if (!client)
-  {
-    CLog::LogF(LOGERROR, "Invalid handler data");
-    return;
-  }
-
-  CServiceBroker::GetPVRManager().EpgContainer().UpdateRequest(client->GetID(), iChannelUid);
+  HandleAddonCallback(__func__, kodiInstance, [&](CPVRClient* client) {
+    CServiceBroker::GetPVRManager().EpgContainer().UpdateRequest(client->GetID(), iChannelUid);
+  });
 }
 
 void CPVRClient::cb_free_demux_packet(void* kodiInstance, DEMUX_PACKET* pPacket)
 {
-  CDVDDemuxUtils::FreeDemuxPacket(static_cast<DemuxPacket*>(pPacket));
+  HandleAddonCallback(__func__, kodiInstance, [&](CPVRClient* client) {
+    CDVDDemuxUtils::FreeDemuxPacket(static_cast<DemuxPacket*>(pPacket));
+  }, true);
 }
 
 DEMUX_PACKET* CPVRClient::cb_allocate_demux_packet(void* kodiInstance, int iDataSize)
 {
-  return CDVDDemuxUtils::AllocateDemuxPacket(iDataSize);
+  DEMUX_PACKET* result = nullptr;
+
+  HandleAddonCallback(__func__, kodiInstance, [&](CPVRClient* client) {
+    result = CDVDDemuxUtils::AllocateDemuxPacket(iDataSize);
+  }, true);
+
+  return result;
 }
 
 void CPVRClient::cb_connection_state_change(void* kodiInstance,
                                             const char* strConnectionString,
                                             PVR_CONNECTION_STATE newState,
                                             const char* strMessage)
 {
-  CPVRClient* client = static_cast<CPVRClient*>(kodiInstance);
-  if (!client || !strConnectionString)
-  {
-    CLog::LogF(LOGERROR, "Invalid handler data");
-    return;
-  }
+  HandleAddonCallback(__func__, kodiInstance, [&](CPVRClient* client) {
+    if (!strConnectionString)
+    {
+      CLog::LogF(LOGERROR, "Invalid callback parameter(s)");
+      return;
+    }
 
-  const PVR_CONNECTION_STATE prevState(client->GetConnectionState());
-  if (prevState == newState)
-    return;
+    const PVR_CONNECTION_STATE prevState(client->GetConnectionState());
+    if (prevState == newState)
+      return;
 
-  CLog::LogFC(LOGDEBUG, LOGPVR,
-              "State for connection '{}' on client '{}' changed from '{}' to '{}'",
-              strConnectionString, client->Name(), prevState, newState);
+    CLog::LogFC(LOGDEBUG, LOGPVR,
+                "State for connection '{}' on client '{}' changed from '{}' to '{}'",
+                strConnectionString, client->Name(), prevState, newState);
 
-  client->SetConnectionState(newState);
+    client->SetConnectionState(newState);
 
-  std::string msg;
-  if (strMessage != nullptr)
-    msg = strMessage;
+    std::string msg;
+    if (strMessage != nullptr)
```

Comment:
`If (!strMessage)` maybe?

Reply:
This was also in the original code, but I can change that.